### DISCUSSION TITLE
import("a") 总是返回 {}

### DISCUSSION THREAD

**Discussion by q962** [xmake-io/xmake#1827] [Q&A]
### 描述问题

import 导入本地lua 文件总是返回 {}
https://github.com/xmake-io/xmake/blob/0ef8c1bf8f4ec10407f298c04d038f0480d9d944/xmake/core/sandbox/sandbox.lua#L272

### 期待的结果

正确返回 lua return 内容

### 相关环境

- xmake版本：2.5.9
- xmake运行平台：
- xmake目标平台：

### 其他信息

file: a.lua
```lua
return {
    a= "a module"
};
````

file: xmake.lua
```lua
on_load
    import("a");
    print(a);
```

看了代码，最终走到 
https://github.com/xmake-io/xmake/blob/0ef8c1bf8f4ec10407f298c04d038f0480d9d944/xmake/core/sandbox/sandbox.lua#L266-L279

errors 是 a.lua 的结果。但是并没有使用它

**Comment by waruqi** [comment]
a.lua 的模块写法不对，外面的脚本域模块是 xmake 抽象定制过得，不走 require，不需要 return 

a.lua
```lua
function _foo()
end

function test()
    _foo()
end

function main()
end
```

其中 main 为入口函数，可选，如果设置，模块a可以直接被调用，例如：

```lua
import("a")
a()
```

其他不带下划线的为 public 模块接口函数，例如 test，

```lua
import("a")
a.test()
```

带下划线前缀的 _foo 是私有函数，模块内部使用，不对外导出，外面不可调用

**Comment by q962** [comment]
把文档完善一下吧

**Comment by waruqi** [comment]
东西太多，时间太少，有时间会逐步完善，但目前的精力不可能一下子全补全，还有大量扩展模块都没空写
